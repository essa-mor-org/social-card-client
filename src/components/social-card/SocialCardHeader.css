:root {
    --Social-header-font: 14px;
    --Social-header-color: #616770;
}

.Social_header{
    display: flex;
    justify-content: space-between;
    padding: 10px;
}

.Social_header_column1{
    display: inline-grid;
    grid-template-columns: auto auto auto auto;
    grid-gap: 5px;
    
}

.Social_header_image_link{
    grid-row-start: 1;
    grid-row-end: 3;
}

.Social_header_image{
    border-radius: 50%;
    width: 40px;
    height: 40px;
}

.Social_header_name{
    color:rgb(54, 88, 153);
    text-decoration-line: none;
    font-size: var(--Social-header-font);
}

.Social_header_name:hover{
    text-decoration-line: underline;
}

.Social_header_time{
    grid-row-start: 2;
    grid-column-start: 2;
    color: var(--Social-header-color);
    text-decoration-line: none;
}

.Social_header_time:hover{
    text-decoration-line: underline;
}

.Social_header_saperator{
    grid-row-start: 2;
    grid-column-start: 3;
    color: var(--Social-header-color);
}

.Social_header_world{
    grid-row-start: 2;
    grid-column-start: 4;
    background-image: url(https://www.facebook.com/rsrc.php/v3/yE/r/lUxJXT2M6Ll.png);
    background-size: 22px 36px;
    background-repeat: no-repeat;
    display: inline-block;
    width: 12px;
    height: 12px;
    background-position-x: 0;
    background-position-y: -22px;
}

.Social_header_toggle{
    background-image: url(https://www.facebook.com/rsrc.php/v3/yJ/r/dskhplJuwiX.png);
    background-repeat: no-repeat;
    background-size: 64px 396px;
    background-position: 0 -196px;
    height: 20px;
    width: 20px;
    display: inline-block;
}