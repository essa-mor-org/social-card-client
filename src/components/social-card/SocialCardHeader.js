import React from 'react';
import PropTypes from 'prop-types';

import './SocialCardHeader.css';

function SocialCardHeader({ image, profileText, time }) {
	return (<div className="Social_header">
		<div className="Social_header_column1">
			<a className="Social_header_image_link" href="#">
				<img className="Social_header_image" src={image} alt=""/>
			</a>
			<a className="Social_header_name" href="#">{profileText}</a>
			<a className="Social_header_time" href="#">{time}</a>
			<span className="Social_header_saperator">.</span>
			<a className="Social_header_world" href="#"></a>
		</div>
		<a className="Social_header_toggle" href="#"></a>
	</div>);
}

SocialCardHeader.propTypes = {
	image: PropTypes.string.isRequired,
	profileText: PropTypes.string.isRequired,
	time: PropTypes.string.isRequired
};

export default SocialCardHeader;