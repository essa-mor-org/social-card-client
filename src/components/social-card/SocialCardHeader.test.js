import React from 'react';
import SocialCardHeader from './SocialCardHeader';
import renderer from 'react-test-renderer';

describe('SocialCardHeader test', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<SocialCardHeader
				image="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg"
				profileText="Vevo"
				time="3 hrs" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});	
});
