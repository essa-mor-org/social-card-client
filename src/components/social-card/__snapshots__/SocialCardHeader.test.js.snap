// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialCardHeader test renders correctly 1`] = `
<div
  className="Social_header"
>
  <div
    className="Social_header_column1"
  >
    <a
      className="Social_header_image_link"
      href="#"
    >
      <img
        alt=""
        className="Social_header_image"
        src="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg"
      />
    </a>
    <a
      className="Social_header_name"
      href="#"
    >
      Vevo
    </a>
    <a
      className="Social_header_time"
      href="#"
    >
      3 hrs
    </a>
    <span
      className="Social_header_saperator"
    >
      .
    </span>
    <a
      className="Social_header_world"
      href="#"
    />
  </div>
  <a
    className="Social_header_toggle"
    href="#"
  />
</div>
`;
